services/browsers/database.go

```diff
@@ -122,3 +124,72 @@ func (db *Database) DeleteBrowser(id string) error {
 	_, err := db.db.Exec("DELETE from browsers WHERE id = ?", id)
 	return err
 }
+
+type BookmarksType string
+
+type Bookmark struct {
+	Url      string `json:"url"`
+	Name     string `json:"name"`
+	ImageUrl string `json:"imageUrl"`
+}
+
+func (db *Database) GetBookmarks() ([]*Bookmark, error) {
+	rows, err := db.db.Query(`SELECT url, name, image_url FROM bookmarks`)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var rst []*Bookmark
+	for rows.Next() {
+		bookmark := &Bookmark{}
+		err := rows.Scan(&bookmark.Url, &bookmark.Name, &bookmark.ImageUrl)
+		if err != nil {
+			return nil, err
+		}
+
+		rst = append(rst, bookmark)
+	}
+
+	return rst, nil
+}
+
+func (db *Database) StoreBookmark(bookmark Bookmark) (Bookmark, error) {
+	insert, err := db.db.Prepare("INSERT OR REPLACE INTO bookmarks (url, name, image_url) VALUES (?, ?, ?)")
+
+	if err != nil {
+		return bookmark, err
+	}
+
+	// Get the right icon
+	finder := besticon.IconFinder{}
+	icons, iconError := finder.FetchIcons(bookmark.Url)
+
+	if iconError == nil {
```

Comment:
you can probably merge the two ifs
`if iconError == nil && len(icons) > 0`
but up to you, either are fine by me